src/sc_generation/generation.rs

```diff
@@ -100,88 +111,115 @@ pub fn generate_calls(
     let mut call_already_prep = false;
     for _ in 0..nb_calls {
         match abi[0].as_str() {
-            "print" => generate_abi_print(&mut rng, &mut calls),
-            "call" => generate_abi_call(
-                &address_sc,
-                &mut calls,
-                &mut preparation_calls,
-                &mut call_already_prep,
-            ),
-            "localCall" => generate_abi_local_call(
+            "abi_set_data" => generate_abi_set_data(&mut rng, &mut calls),
```

Comment:
I feel that injecting rng makes it less readable.
Some times it's injected some time not, while I feel that it should be always or never.
What about removing the parameter and instantiating it if needed in the function itself?

Reply:
Done